crates/dyn-abi/src/ext/abi.rs

```diff
@@ -92,8 +92,8 @@ impl JsonAbiExt for Constructor {
     }
 
     #[inline]
-    fn decode_input(&self, data: &[u8]) -> Result<Vec<DynSolValue>> {
-        decode(data, &self.inputs)
+    fn decode_input(&self, data: &[u8], validate: bool) -> Result<Vec<DynSolValue>> {
```

Comment:
What's our general stance on functions which have `validate: bool`? When is it OK to not validate?

Reply:
`validate` set to `true` is the same as calling the `_whole` version of the function, which I didn't know existed prior to this